client/transport/grpc/src/main/java/io/a2a/client/transport/grpc/GrpcTransport.java

```diff
@@ -59,9 +61,12 @@ public EventKind sendMessage(MessageSendParams request, ClientCallContext contex
         checkNotNullParam("request", request);
 
         SendMessageRequest sendMessageRequest = createGrpcSendMessageRequest(request, context);
+        Metadata metadata = createGrpcMetadata(context);
 
         try {
-            SendMessageResponse response = blockingStub.sendMessage(sendMessageRequest);
+            // Create a stub with metadata attached
+            A2AServiceBlockingV2Stub stubWithMetadata = blockingStub.withInterceptors(MetadataUtils.newAttachHeadersInterceptor(metadata));
```

Comment:
Just to check, the metadata is only being used for sendMessage and sendStreamingMessage. Is that intentional? Should it be used for all requests?